### PR TITLE
Matrix batches

### PR DESCRIPTION
**Pull request by tgymnich**
This PR would enable lu factorization in batches. However the MPSKernel only ever runs once for LU decomposition, trying the same for MatMul works just fine.

### CODE REVIEW

**Comment by tgymnich** [comment]
Would be nice to direct something like:
`eachslice(X; dims=3) .* eachslice(Y; dims=3)` or `@einsum C[i,j,b] := A[i,k,b] * B[k,j,b]` to the batch MPS matmul kernel.

**Comment by maleadt** [comment]
> `eachslice(X; dims=3) .* eachslice(Y; dims=3)` or `@einsum C[i,j,b] := A[i,k,b] * B[k,j,b]` to the batch MPS matmul kernel

Yeah, I'm surprised we haven't come up with anything convenient yet. There's https://github.com/FluxML/NNlib.jl/blob/1b30040fabadd41efa0d9dde5841b90f9f85cf2d/src/batched/batchedmul.jl#L4-L54, but that would require an NNlib.jl dependency.

File: lib/mps/matrix.jl [hunk 1]
```diff
@@ -173,6 +173,8 @@ function encode!(cmdbuf::MTLCommandBuffer, matmul::MPSMatrixMultiplication, left
                                                resultMatrix:result::id{MPSMatrix}]::Nothing
 end
 
+const MtlMatrixBatch{T,S} = MtlArray{T,3,S}
```
Comment (maleadt) [inline lib/mps/matrix.jl:176]: Unused alias?

Comment (tgymnich) [reply lib/mps/matrix.jl:176]:
```suggestion
```

Comment (tgymnich) [reply lib/mps/matrix.jl]: removed

**Pull request merged**